Introducing multiplatform ViewModels + Presenters

pmieszal opened the pull request:
- general ViewModel interface
- fixed KDate package
- KUUID for native uuids

Implements: #7

pmieszal committed:
- introducing ViewModels + Presenters
- general ViewModel interface
- fixed KDate package
- KUUID for native uuids

pmieszal committed:
- redundant null init for name property
- redundant optional treating for date property
- private set for events list

pmieszal committed:
- private view models fields

pmieszal committed:
whitespace

mateuszkwiecinski commented on atter-logic/src/commonMain/kotlin/com.attendance.checker/events/viewModels/EventEditViewModel.kt:
```diff
@@ -0,0 +1,64 @@
+        presenter.displayDate(date.isoDate)
+    }
+
+    fun didChangeName(name: String) {
```
I'd recommend to move this logic to property setter 
```
var date : KDate
   set(value) { 
       field = value
       refreshSaveButtonStatus()
  }
```

pmieszal replied:
Good point.

mateuszkwiecinski commented on atter-logic/src/commonMain/kotlin/com.attendance.checker/events/viewModels/EventEditViewModel.kt:
```diff
@@ -0,0 +1,64 @@
+import com.attendance.checker.shared.viewModels.ViewModel
+import com.attendance.checker.uuid.KUUID
+
+class EventEditViewModel constructor(private val presenter: EventEditPresenter,
```
Did you try to use: https://github.com/florent37/Multiplatform-LiveData to avoid having a`*Presenter` class? 
(Which by the way I'd call `EventEditView`) :P

pmieszal replied:
Don't work (as discussed on slack)

pmieszal committed:
- EventEditPresented displaying KDate
- EventEditViewModel refresh save button on name/date setter
- KDate redundant date property

pmieszal committed:
Merge branch 'master' into feature/view-models

# Conflicts:
#	atter-logic/src/androidMain/kotlin/com/attendance/checker/date/KDate.kt
#	atter-logic/src/commonMain/kotlin/com.attendance.checker/events/Event.kt
#	atter-logic/src/commonMain/kotlin/com.attendance.checker/events/GetEvents.kt
#	atter-logic/src/commonMain/kotlin/com.attendance.checker/events/SaveEvent.kt
#	atter-logic/src/commonMain/kotlin/com/attendance/checker/events/Event.kt
#	atter-logic/src/commonMain/kotlin/com/attendance/checker/events/GetEvents.kt
#	atter-logic/src/commonMain/kotlin/com/attendance/checker/events/SaveEvent.kt
#	atter-logic/src/commonMain/kotlin/com/attendance/checker/events/models/Event.kt
#	atter-logic/src/commonMain/kotlin/com/attendance/checker/events/useCases/GetEvents.kt
#	atter-logic/src/commonMain/kotlin/com/attendance/checker/events/useCases/SaveEvent.kt
#	atter-logic/src/iosMain/kotlin/com.attendance.checker/date/KDate.kt

pmieszal committed:
packages + new lines

pmieszal committed:
MainActivity imports
